motoman_driver/MotoPlus/MotionServer.c

```diff
@@ -268,7 +271,14 @@ void Ros_MotionServer_WaitForSimpleMsg(Controller* controller, int connectionInd
         			expectedSize = minSize + sizeof(SmBodyMotoMotionReply);
 	    			break;
 				case ROS_MSG_MOTO_JOINT_TRAJ_PT_FULL_EX:
-					expectedSize = minSize + sizeof(SmBodyJointTrajPtFullEx);
+					//Don't require the user to send data for non-existant control groups
+					if (byteSize >= (minSize + sizeof(int))) //make sure I can at least get to [numberOfGroups] field

```

Comment:
I modified it to work with either format... the full size, or the truncated size with data for only limited groups.  Changes to the ROS side are not required.
The first user who tested this for a linear track was sending abbreviated messages by mistake... so I decided to make the driver accept either format.